doc/crypto/api/ops/kdf.rst

```diff
@@ -153,6 +153,98 @@ Key derivation algorithms
         | `PSA_KEY_TYPE_DERIVE` (for the pseudorandom key)
         | `PSA_KEY_TYPE_RAW_DATA` (for the info string)
 
+.. macro:: PSA_ALG_SP800_108_COUNTER_HMAC
+    :definition: /* specification-defined value */
+
+    .. summary::
+        Macro to build a NIST SP 800-108 conformant, counter-mode KDF algorithm based on HMAC.
+
+    .. param:: hash_alg
+        A hash algorithm: a value of type `psa_algorithm_t` such that :code:`PSA_ALG_IS_HASH(hash_alg)` is true.
+
+    .. return::
+        The corresponding key derivation algorithm. For example, the counter-mode KDF using HMAC-SHA-256 is :code:`PSA_ALG_SP800_108_COUNTER_HMAC(PSA_ALG_SHA_256)`.
+
+        Unspecified if ``hash_alg`` is not a supported hash algorithm.
+
+    This is the HMAC-based, counter mode key derivation function specified by :cite-title:`SP800-108`, §4.1.
+
+    This key derivation algorithm uses the following inputs, in any order:
```

Comment:
Redesigning the KDF operation to take the secret input key as part of the setup function would also simplify the design of the KDF crypto driver API and driver implementations considerably. Therefore we think it is worthwhile.

Reply:
> Redesigning the KDF operation to take the secret input key as part of the setup function would also simplify the design of the KDF crypto driver API and driver implementations considerably. Therefore we think it is worthwhile.

The question is whether the benefit is worth the pain of breaking applications that do not migrate to the new API. This requires more detailed analysis of the use cases where this matters for the implementation, the cost of supporting the existing AND a new API in an implementation, what is expected for use cases where this API is not used (or cannot be used), etc.

Reply:
As an implementer, I would prefer to wait until we have more API-breaking changes, and group them in a 2.0 API.

In particular, I don't want to make a 2.0 API until we've settled on an API for PQC. We'll see if that requires significant changes that warrant a 2.0.

Reply:
Ok. Introducing this can wait for a later update. Although we can introduce a parallel API prior to the breaking change - but I think we need to consider the migration story before we do that.